### PR TITLE
Implement repo priorities.

### PR DESCRIPTION
**Pull request by nguyen-phillip**
This adds an integer priority value to repo entries in repo config files. If not specified, a default priority of 500 is used.

These priorities are associated with package versions in the map returned by client.AvailableVersions. When looking for the latest version, client.FindRepoLatest takes priorities into account such that a package version from a higher priority repo will always compare greater than any package version from a lower priority repo.

During "googet install" the highest priority version will be installed if the package is not already installed, otherwise the install logic remains the same if the package is already installed.

During "googet update" versions with priority greater than the default priority of 500 will take precedence over installed versions. The highest priority version will be installed even if it means downgrading the installed version in order to facilitate rollbacks.

### CODE REVIEW

File: goolib/goospec_test.go [hunk 1]
```diff
@@ -205,6 +206,38 @@ func TestBadCompare(t *testing.T) {
...
 
+func TestComparePriorityVersion(t *testing.T) {
+	for _, tc := range []struct {
+		p1   int
+		v1   string
+		p2   int
+		v2   string
+		want int
+	}{
+		{500, "1.2.3@1", 500, "1.2.3@2", -1},
```
Comment (adjackura) [inline goolib/goospec_test.go:217]: Can you add a  short description to each test case and print that when "got != tc.want"

**Review by adjackura — CHANGES_REQUESTED** [review]

File: goolib/goospec_test.go (hunk 1 above)
Comment (nguyen-phillip) [reply goolib/goospec_test.go]: Done.

**Comment by nguyen-phillip** [comment]
PTAL

**Review by adjackura — APPROVED** [review]

**Comment by adjackura** [comment]
Can you rebase with the new repo changes (should be minor)

**Comment by nguyen-phillip** [comment]
> Can you rebase with the new repo changes (should be minor)

Done. Sorry for the long delay, have been out on leave. Thanks for the review!

**Comment by nguyen-phillip** [comment]
PTAL: changed the updates logic to prevent unnecessary reinstalls and added some tests.

**Review by adjackura — APPROVED** [review]

**Pull request merged**